department.rb

```diff
@@ -0,0 +1,33 @@
+class Department
+  attr_accessor :employees, :dept_name
+
+  def initialize(dept_name)
+    @dept_name = dept_name
+    @employees = []
+  end
+
+  def add_employee(employee)
+    @employees << employee
+  end
+
+  def has_employee_with_name?(name)
+    @employees.any? {|e| e.name == name}
+  end
+
+  def total_salary
+    @employees.reduce(0.0) {|sum, employee| sum += employee.salary}
+  end
+
+  def department_raise(amount)
+    @amount = amount
+    until @amount == 0
```

Comment:
Oh, wow.  I hadn't thought about doing it this way, but I suppose that it works on average.  You definitely want to be the first person in this array!